tests/test_parsing.js

```diff
@@ -1561,3 +1599,26 @@ function createParenthesizedNode(value) {
     value: value,
   };
 }
+
+/**
+ * @template T
```

Comment:
same. I think that `T={type: string, name: string }` (or maybe a more specific type)

Reply:
anyway, this is test code, so I don't know if the overhead of jsdoc types is worth it here.

Reply:
Well, in&nbsp;this&nbsp;case, `T`&nbsp;can&nbsp;be&nbsp;any&nbsp;subtype of&nbsp;`Node`.

Reply:
In that case I think it should be `@template {Node} T` -- as is `T` will bind to any type, so you could get a return type `{ type: NodeType, name: string | boolean | undefined }` or something crazy like that.